pkg/nodeconfig/nodeconfig.go

```diff
@@ -111,6 +114,19 @@ func discoverKubeAPIServerEndpoint() (string, error) {
 	return host.Status.APIServerInternalURL, nil
 }
 
+// getClusterDNSIP return 10th IP for the given CIDR IP range
```

Comment:
Can we add a bit explaining where this logic comes from?
`This is a widespread convention set by upstream Kubernetes (kubeadm)`

Not sure if it would be better as a code comment or in the commit+PR messages, but should be somewhere imo